### PR TITLE
Feature/add disable resource alloc destroy

### PR DESCRIPTION
**Pull request by talekhinezh**
* Add header flag to allow resource allocation/destruction to be disabled

### CODE REVIEW

**Review by siy — APPROVED** [review]
LGTM

File: radixdlt-engine/src/main/java/com/radixdlt/atommodel/tokens/scrypt/TokensConstraintScryptV3.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+				}
+
+				return new Authorization(PermissionLevel.USER, (r, c) -> {
+					var tokenDef = (TokenResource) r.loadAddr(null, u.getResourceAddr())
+						.orElseThrow(() -> new AuthorizationException("Invalid token address: " + u.getResourceAddr()));
+					tokenDef.verifyMintAuthorization(c.key());
+				});
+			},
+			(s, u, c, r) -> {
+				c.verifyCanDestroyResources();
```
Comment (iamyulong) [inline radixdlt-engine/src/main/java/com/radixdlt/atommodel/tokens/scrypt/TokensConstraintScryptV3.java:123]: Can we rename `verifyCanDestroyResources` to indicate that it controls both allocate and deallocate  resources?

**Review by iamyulong — COMMENTED** [review]
LGTM

**Review by LukasGasior1 — APPROVED** [review]
LGTM

**Pull request merged**